src/vcu/src/controls.c

```diff
@@ -105,15 +118,72 @@ void execute_controls(void) {
     int32_t limited_torque;
     if (dash_limited_torque < min_sensor_torque) {
       limited_torque = dash_limited_torque;
-    }
-    else {
+    } else {
       limited_torque = min_sensor_torque;
     }
 
     torque_command = limited_torque;
+    bool torque_limited = limited_torque < controls_monitoring.raw_torque;
+  
+    if (lc_settings.using_launch_ctrl && lc_state != DONE && !torque_limited) {
+      launch_ctrl_entered = true;
+
+      // We shouldn't be braking in launch control, so reset regen valve
+      set_brake_valve(false);
+      uint32_t front_wheel_speed = get_front_wheel_speed();
+
+      // Mini FSM
+      switch (lc_state) {
+        case BEFORE:
+          sendSpeedCmdMsg(0, 0);
+
+          // Transition
+          if (pedalbox_min(accel) > LC_ACCEL_BEGIN) {
+            lc_state = SPEEDING_UP;
+            printf("[LAUNCH CONTROL] SPEEDING UP STATE ENTERED\r\n");
+          }
+          break;
+        case SPEEDING_UP:
+          if (1000 > torque_command) sendSpeedCmdMsg(500, torque_command);
```

Comment:
Why 1000? What does this variable indicate? Why 500? What does this variable indicate.